src/views/common/Animation/AnimationControls/useDraggableScrubber.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useCallback, useMemo, useRef } from 'react'
```

Comment:
This guy uses the throttler from the rateLimiters, while also exposing some logic that determines whether and how to handle initiating and concluding the overall scrub action.